File: codes/perl/inheritance/multiple/multiple1.pl
package Person;

sub new {
    my $class = shift;
    my $self = {};
    bless $self, $class;
    print "I am a person\n";
    return $self;
}

1;

package Machine;

sub new {
    my $class = shift;
    my $self = {};
    bless $self, $class;
    print "I am a machine\n";
    return $self;
}

1;

package Android;
our @ISA = qw(Machine Person);

sub new {
    my $class = shift;
    my $self = $class->Person::new();
    my $self2 = $class->Machine::new();
    $self = { %$self, %$self2};
    bless $self, $class;
    print "So, I am an android\n";
    return $self;
}

1;

package Cyborg;
our @ISA = qw(Person Machine);

sub new {
    my $class = shift;
    my $self = $class->Machine::new();
    my $self2 = $class->Person::new();
    $self = { %$self, %$self2};
    bless $self, $class;
    print "So, I am a cyborg\n";
    return $self;
}

1;

$p = Person->new();
print "=====================\n";
$m = Machine->new();
print "=====================\n";
$a = Android->new();
print "=====================\n";
$c = Cyborg->new();
print "=====================\n";

File: codes/java/src/inheritance/multiple/Multiple1.java
package inheritance.multiple;

public class Multiple1 {

    public static void main(String[] args) {
        Person p = new Person();
        //p.info();
        System.out.println("=====================");
        Machine m = new Machine();
        //m.info();
        System.out.println("=====================");
        Android a = new Android();
        //a.info();
        System.out.println("=====================");
        Cyborg c = new Cyborg();
        //c.info();
        System.out.println("=====================");
    }

}

interface HumanBehaviour {
    default void info() {
        System.out.println("I am a person");
    }
}

interface MachineBehaviour {
    default void info() {
        System.out.println("I am a machine");
    }
}

class Person implements HumanBehaviour {
    public Person(){
        info();
    }
}

class Machine implements MachineBehaviour {
    public Machine(){
        info();
    }
}

class Android extends Machine implements HumanBehaviour {
    public void info() {
        super.info();
        HumanBehaviour.super.info();
        System.out.println("So, I am an android");
    }
}

class Cyborg extends Person implements MachineBehaviour {
    public void info() {
        super.info();
        MachineBehaviour.super.info();
        System.out.println("So, I am a cyborg");
    }
}
